mongo/bson.nim

```diff
@@ -767,10 +767,32 @@ proc newBsonDocument*(bytes: string): Bson =
 
 ## Serialization/deserialization ext
 
+# ------------- type: Serializer/Deserializer -------------------#
+
+type
+  BsonSerializer* = concept s
+    ## type that can serialize itself to a bson object (implemented by all primitive type)
+    s.toBson() is Bson
+  BsonDeserializer* = concept d
+    ## type that can deserialize a bson type to itself
+    Bson.to(d)
+
 template dbKey*(name: string) {.pragma.}
+template omitempty*() {.pragma.}
+
+proc toBson*(d: DateTime) : Bson = ## Serializer for DateTime type
```

Comment:
style: Documentation comments should go on their own line when they are the first statement of a procedure body.